test/unit/account_test.rb

```diff
@@ -72,11 +72,9 @@ def setup
     master = master_account
     buyer = FactoryBot.create(:simple_buyer, provider_account: master)
 
-    assert_includes master.buyer_account_ids, buyer.id
-    assert_includes master.buyer_account_ids, master.id
-
-    assert_includes master.buyer_accounts.not_master.ids, buyer.id
-    assert_not_includes master.buyer_accounts.not_master.ids, master.id
+    assert_includes Account.all.ids, master.id
+    assert_includes Account.all.not_master.ids, buyer.id
+    assert_not_includes Account.all.not_master.ids, master.id
```

Comment:
I understand the old `assert_includes master.buyer_account_ids, master.id` isn't valid anymore, but why replacing `buyer_accounts` by `Account.all` in the other assertions? It seems to me we should test `buyer_accounts` since it's precisely what changed.

Reply:
I believe the test intention is to test the `not master` scope. To be a good test, the originally used scope should include the master account and once `not_master` is applied, then it should not include master account anymore.

In that sense `buyer_account_ids` is not a good scope to test `not_master` with. Where we test `buyer_accounts` in another test case specifically for it.